pom.xml

```diff
@@ -198,24 +234,24 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-text</artifactId>
             <version>1.9</version>
-            <scope>provided</scope>
+            <scope>compile</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
             <version>3.12.0</version>
-            <scope>provided</scope>
+            <scope>compile</scope>
         </dependency>
         <dependency>
             <groupId>org.osgi</groupId>
```

Comment:
Replacing `osgi.core` by the individual spec chapter dependencies is hard, as most dependencies transitively just declare a dependency on `osgi.core` and not the individual spec chapters, therefore I would tend to leave it like that although most probably not all packages/spec chapters are really necessary here.